components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java

```diff
@@ -846,24 +820,10 @@ public Response apisApiIdTopicsGet(String apiId, String xWSO2Tenant, MessageCont
     @Override
     public Response apisApiIdUserRatingPut(String id, RatingDTO body, String xWSO2Tenant,
             MessageContext messageContext) {
-        String requestedTenantDomain = RestApiUtil.getRequestedTenantDomain(xWSO2Tenant);
         try {
             int rating = 0;
             String username = RestApiCommonUtil.getLoggedInUsername();
-            if (!APIUtil.isTenantAvailable(requestedTenantDomain)) {
-                RestApiUtil.handleBadRequest("Provided tenant domain '" + xWSO2Tenant + "' is invalid",
-                        ExceptionCodes.INVALID_TENANT.getErrorCode(), log);
-            }
             APIConsumer apiConsumer = RestApiCommonUtil.getConsumer(username);
-            //this will fail if user doesn't have access to the API or the API does not exist
-            ApiTypeWrapper apiTypeWrapper = apiConsumer.getAPIorAPIProductByUUID(id, requestedTenantDomain);
```

Comment:
We will hv to keep this check for api visibility related checks. Without this, users who do not hv visibility permission can edit it. Check wether it is handled from different place

Reply:
we could add a similar check done in https://github.com/wso2/carbon-apimgt/blob/master-org-refactor/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java#L6117 here as well

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/10512/commits/2bbd75f03d2e35bf306fa99a5e37c4268b9e3b9a